File: server/src/main/java/com/github/genraven/genesys/domain/common/Action.java
package com.github.genraven.genesys.domain.common;

import com.github.genraven.genesys.domain.enums.Difficulty;
import com.github.genraven.genesys.domain.enums.RangeBand;
import com.github.genraven.genesys.domain.actor.ActorSkill;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
public class Action {

    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private ActorSkill skill;

    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private Difficulty difficulty;

    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private ActorSkill opposedSkill;

    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private RangeBand rangeBand = RangeBand.ENGAGED;
}

File: client/src/models/campaign/encounter/Action.ts
import { ActorSkill } from "../../actor/Actor"
import {CharacteristicType, Difficulty, RangeBand, SkillType} from "../../../api/model";

export default interface Action {
    skill: ActorSkill
    difficulty: Difficulty
    opposedSkill: ActorSkill
    rangeBand: RangeBand
}

export const getEmptyAction = (): Action => ({
  skill: {
      ranks: 0,
      id: "",
      characteristic: CharacteristicType.Brawn,
      type: SkillType.General,
      name: "",
      initiative: false
  },
  difficulty: Difficulty.Average,
  opposedSkill: {
      ranks: 0,
      id: "",
      characteristic: CharacteristicType.Brawn,
      type: SkillType.General,
      name: "",
      initiative: false
  },
  rangeBand: RangeBand.Engaged,
});
